kratos/includes/kratos_filesystem.h

```diff
@@ -7,39 +7,48 @@
 //  License:		 BSD License
 //					 Kratos default license: kratos/license.txt
 //
-//  Main authors:    Philipp Bucher
+//  Main authors:    Philipp Bucher (https://github.com/philbucher)
 //
 
 #if !defined(KRATOS_FILESYSTEM)
 #define KRATOS_FILESYSTEM
 
+// System includes
+#include <string>
+#include <vector>
+
+// External includes
+
 // Project includes
-#include "utilities/check_filesystem_availability.h" // has to be included before using "KRATOS_FILESYSTEM_AVAILABLE"
-
-// System / External includes
-// std::filesystem is part of C++17.
-// Until Kratos moves C++17, we use the ghc-filesystem library, which is a C++11 version of std::filesystem
-#if defined(KRATOS_FILESYSTEM_AVAILABLE)
-    #include <filesystem>
-#else
-    //---------------------------------------------------------------------------------------
-    // fs_fwd.hpp - The forwarding header for the header/implementation seperated usage of
-    //              ghc::filesystem.
-    // This file can be include at any place, where ghc::filesystem api is needed while
-    // not bleeding implementation details (e.g. system includes) into the global namespace,
-    // as long as one cpp includes fs_impl.hpp to deliver the matching implementations.
-    //---------------------------------------------------------------------------------------
-    #include "ghc/fs_fwd.hpp"
-#endif
+#include "includes/define.h"
+
 
 namespace Kratos {
+// Functions that provide some basic functionalities that are provided by std::filesystem (part of C++17)
+// please check the documentation of std::filesystem for the function documentation
+namespace FileSystem {
+
+bool KRATOS_API(KRATOS_CORE) Exists(const std::string& rPath);
```

Comment:
I thought we were planning to use the same names as the filesystem library in order to easy replace in C++17. Otherwise, we already had the OSUtilities...

Reply:
I know, I am still unsure
I went back and forth when implementing

My thought was that we could enhance the Namespace with our own functions (like `JoinPaths`)

But now I am thinking that I could split it into `Kratos::filesystem` (to be replaced by `std::filesystem`) and `Kratos::FileSystemExtensions` which contains our own functions

What do you think?

Reply:
Ok for me, in any case if the problem is comming from the linking in windows due to the lack of source file, maybe this solution is overengineered

Reply:
But this is not the only header-only lib we use
Now the concept is the same with the json-lib :confused: 

Lets see, I haven't given up yet

The biggest issue is the deletion of a non-empty folder, which would be super handy to have